lib/queue.js

```diff
@@ -1079,12 +1079,25 @@ Queue.prototype.getJobs = function(queueType, type, asc, start, end){
 
       break;
     case 'ZSET':
-      jobs = asc ? this.client.zrange(key, start, end) : this.client.zrevrange(key, start, end);
+      jobs = (asc ? this.client.zrange(key, start, end, 'WITHSCORES') : this.client.zrevrange(key, start, end, 'WITHSCORES'))
+        .then(function(result){
+          var jobIds = [];
+          for(var i=0; i<result.length; i++){
+            if(i&1){
+              timestamps[result[i-1]] = result[i];
+            }else{
+              jobIds.push(result[i]);
+            }
+          }
+          return jobIds;
+        });
       break;
   }
 
   return jobs.then(function(jobIds){
-    var jobsFromId = jobIds.map(_this.getJobFromId);
+    var jobsFromId = jobIds.map(function(jobId){
+      return Job.fromId(_this, jobId, timestamps[jobId]);
```

Comment:
@manast it appears the new `timestamp` argument to `Job.fromId()` is not referenced. What was the intention here?